messages.rb

```diff
@@ -505,6 +506,84 @@ def send_diff(event)
   event << "Score changes on #{current.format_name} since #{since}:\n```#{diff}```"
 end
 
+def do_analysis(scores, rank)
+  run = scores.download_scores(rank)
+  if !run.nil?
+    player = run['user_name']
+    replay_id = run['replay_id'].to_s
+    score = "%.3f" % [run['score'].to_f / 1000]
+    analysis = scores.analyze_replay(replay_id)
+    gold = "%.0f" % [((run['score'].to_f / 1000) + (analysis.size.to_f / 60) - 90) / 2]
+    result = {'player' => player, 'scores' => scores.format_name, 'rank' => rank, 'score' => score, 'analysis' => analysis, 'gold' => gold}
+    return result
+  else
+    return nil
+  end
+end
+
+def send_analysis(event)
+  msg = event.content
+  scores = parse_level_or_episode(msg)
+  ranks = msg.scan(/\s+([0-9][0-9]?)/).map{ |r| r[0].to_i }.reject{ |r| r < 0 || r > 19 }
+  if ranks.empty? then ranks.push(0) end
+  analysis = ranks.map{ |rank| do_analysis(scores, rank) }.compact
+  length = analysis.map{ |a| a['analysis'].size }.max
+  padding = Math.log(length, 10).to_i + 1
+  table_header = " " * padding + "|" + "JRL|" * analysis.size
+  separation = "-" * table_header.size
+
+  # 3 types of result formatting, only 2 being used.
+
+  raw_result = analysis.map{ |a|
+    a['analysis'].map{ |b|
+      [b % 2 == 1, b / 2 % 2 == 1, b / 4 % 2 == 1]
+    }.map{ |f|
+      frame = ""
+      if f[0] then frame.concat("j") end
+      if f[1] then frame.concat("r") end
+      if f[2] then frame.concat("l") end
+      frame
+    }.join(".")
+  }.join("\n\n")
+
+  table_result = analysis.map{ |a|
```

Comment:
There are a couple super long lines in this block :P I generally prefer to linebreak lengthy method chains at ~80 characters which is fairly standard.

Reply:
Shortened ;)

Reply:
So I would normally just accept this if it was the only thing in here... and it makes me feel like a horrible pedant... but since you're going back in here for a second or two anyways, for chained methods like `flatten(1).transpose.each_with_index.map { ...}` I prefer each call on its own line. Makes it easier to see each transformation step-by-step.

Reply:
Heh don't worry, I agree it's clearer that way.